python/dgl/graph_index.py

```diff
@@ -703,12 +745,18 @@ def from_scipy_sparse_matrix(self, adj):
         ----------
         adj : scipy sparse matrix
         """
-        self.clear()
-        self.add_nodes(adj.shape[0])
+        assert isinstance(adj, (scipy.sparse.csr_matrix, scipy.sparse.coo_matrix)), \
+                "The input matrix has to be a SciPy sparse matrix."
+        if not self.is_readonly():
+            self.clear()
+        # what if the adj matrix isn't symmetric.
+        num_nodes = max(adj.shape[0], adj.shape[1])
```

Comment:
I think we should raise error when the given matrix is not square.

Reply:
edge list input handles a similar case.
https://github.com/dmlc/dgl/pull/342/files/3abbf81f267a61a3fcc1a69f55aaaa88c29a6f94#diff-57dab646fc76271b216df6298141ebbeR776

Reply:
I tried to fix it and realized that we may not need to fix it. Think about edge list input. It needs to handle a similar case (max source id might be smaller or larger than the destination id).
https://github.com/dmlc/dgl/pull/342/files/3abbf81f267a61a3fcc1a69f55aaaa88c29a6f94#diff-57dab646fc76271b216df6298141ebbeR776